Formula/php.rb

```diff
@@ -0,0 +1,333 @@
+
+    args = %W[
+      --prefix=#{prefix}
+      --localstatedir=#{var}
+      --sysconfdir=#{config_path}
+      --with-config-file-path=#{config_path}
+      --with-config-file-scan-dir=#{config_path}/conf.d
+      --mandir=#{man}
+      --enable-bcmath
+      --enable-calendar
+      --enable-dba
+      --enable-dtrace
+      --enable-exif
+      --enable-ftp
+      --enable-fpm
+      --enable-gd-native-ttf
+      --enable-intl
+      --enable-mbregex
+      --enable-mbstring
+      --enable-mysqlnd
+      --enable-pcntl
+      --enable-phpdbg
+      --enable-phpdbg-webhelper
+      --enable-shmop
+      --enable-soap
+      --enable-sockets
+      --enable-sysvmsg
+      --enable-sysvsem
+      --enable-sysvshm
+      --enable-wddx
+      --enable-zip
+      --libexecdir=#{libexec}
+      --with-apxs2=#{Formula["httpd"].opt_bin}/apxs
+      --with-bz2=/usr
+      --with-enchant=#{Formula["enchant"].opt_prefix}
+      --with-freetype-dir=#{Formula["freetype"].opt_prefix}
+      --with-gmp=#{Formula["gmp"].opt_prefix}
+      --with-gd
+      --with-gettext=#{Formula["gettext"].opt_prefix}
+      --with-fpm-user=_www
+      --with-fpm-group=_www
+      --with-iconv-dir=/usr
+      --with-icu-dir=#{Formula["icu4c"].opt_prefix}
+      --with-jpeg-dir=#{Formula["jpeg"].opt_prefix}
+      --with-kerberos=/usr
+      --with-ldap=shared
+      --with-ldap-sasl=/usr
+      --with-libxml-dir=/usr
+      --with-imap=shared,#{Formula["imap-uw"].opt_prefix}
+      --with-imap-ssl=#{Formula["openssl"].opt_prefix}
+      --with-libzip=#{Formula["libzip"].opt_prefix}
+      --with-mhash
+      --with-mysql-sock=/tmp/mysql.sock
+      --with-mysqli=mysqlnd
+      --with-pdo-mysql=mysqlnd
+      --with-pdo-odbc=unixODBC,#{Formula["unixodbc"].opt_prefix}
+      --with-ndbm=/usr
+      --with-openssl=#{Formula["openssl"].opt_prefix}
+      --with-pdo-dblib=#{Formula["freetds"].opt_prefix}
+      --with-pdo-pgsql=#{Formula["libpq"].opt_prefix}
+      --with-pgsql=#{Formula["libpq"].opt_prefix}
+      --with-png-dir=#{Formula["libpng"].opt_prefix}
+      --with-pspell=#{Formula["aspell"].opt_prefix}
+      --with-snmp
+      --with-tidy=shared,#{Formula["tidy-html5"].opt_prefix}
+      --with-unixODBC=#{Formula["unixodbc"].opt_prefix}
+      --with-webp-dir=#{Formula["webp"].opt_prefix}
+      --with-pic
+      --with-xmlrpc
+      --with-zlib=/usr
+      --with-libedit
+      --with-xsl=/usr
+    ]
+
+    if MacOS.version < :lion
+      args << "--with-curl=#{Formula["curl"].opt_prefix}"
+    else
+      args << "--with-curl"
+    end
+
+    args << "--enable-maintainer-zts" if build.with? "thread-safety"
+
+    if build.devel?
+      args += %W[
+        --with-password-argon2=#{Formula["argon2"].opt_prefix}
+        --with-sodium=#{Formula["libsodium"].opt_prefix}
+      ]
+    else
+      args << "--with-mcrypt=shared,#{Formula["mcrypt"].opt_prefix}"
+    end
+
+    system "./configure", *args
+
+    inreplace "Makefile" do |s|
```

Comment:
This looks very fragile. Can we get some configure options for this upstream?

Reply:
I'm not sure this is no longer necessary. This replaces the path to whatever 
```
➜  php-src git:(master) ✗ apxs -q LIBEXECDIR
/usr/local/lib/httpd/modules
```
outputs which now is the same as we are changing it into ie. `LIBEXECDIR='#{lib}/httpd/modules'`

Reply:
The `-a` section remocal is only necessay if `apxs -q SYSCONFDIR` returns false which shouldn't happen if your system is brewed correctly.

Reply:
Than second section in this block is however there to sort out the system vs brew libraries that was identified earlier so I'm not sure that can go unless we get upstream patched.

Reply:
>I'm not sure this is no longer necessary. _...(LIBEXECDIR)_

That gsub is necessary, otherwise the formula will attempt to install to that location which will be linked to the httpd cellar. Also, I believe we want to keep all installed files in the php cellar.

> The `-a` section removal is only necessary if `apxs -q SYSCONFDIR` returns false which shouldn't happen if your system is brewed correctly.

I believe the opposite is true. If the `SYSCONFDIR` _is_ set (and it is in httpd), php will attempt to modify the httpd default config file (httpd.conf) on install. I believe it is desirable to prevent the install from touching the config file (the caveats already call out what needs to be added the config), or at the very least prevent it from enabling a new module by replacing `-a` with `-A`.

> second section in this block is however there to sort out the system vs brew libraries that was identified earlier so I'm not sure that can go unless we get upstream patched.

Correct, and I don't personally find it all that fragile. It's going to be hard to get any upstream attention on this as I'm not aware if there's a standard way to prioritize certain library directories in the configure/make process. This is very macOS/Homebrew specific.

Reply:
I don't want to be revisiting `inreplace` string replacements every time a new version of this file comes out.

Reply:
The `INSTALL_IT` one should be pretty stable. The meat of it hasn't changed in 15 years: https://github.com/php/php-src/blame/PHP-7.1/sapi/apache2handler/config.m4#L75-L80